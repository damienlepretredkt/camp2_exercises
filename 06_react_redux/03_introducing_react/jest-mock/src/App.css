.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #324279;
  height: 120px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}


.temp-array {
  display: flex;
  justify-content: center;
}

table thead tr td {
  font-weight: bold;

}


.paper {
  width: '100%';
  marginTop: theme.spacing.unit * 3;
  overflowX: 'auto';
}

.table {
  minWidth: 700,
}
