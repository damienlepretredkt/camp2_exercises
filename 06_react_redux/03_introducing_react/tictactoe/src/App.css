.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

.center-table {
  display: flex;
  justify-content: center;
  margin: 8px 0px 8px 0px;
}

tr {
  border-style: solid;
}

td {
  border-style: solid;
  height: 36px;
  width: 36px;
  font-weight: bold;
  font-size: 18px;
}
