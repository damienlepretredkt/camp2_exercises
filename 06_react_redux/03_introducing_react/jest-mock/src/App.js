import React, { Component } from 'react';
import logo from './sunshine.png';
import './App.css';
import loadWeather from './services/weather'

import ZipCodeForm from './ZipcodeForm';
import WeatherList from './WeatherList';


class App extends Component {

  constructor(props) {
    super(props)
    this.state = {
      zipcode: '',
      city: '',
      weathers: []
    }
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">{this.state.city}</h1>
        </header>

        <ZipCodeForm
          handleChange={this.handleChange}
          handleSubmit={this.handleSubmit}
          zipcode={this.state.zipcode}>
        </ZipCodeForm>

        <WeatherList
          weathers={this.state.weathers}>
        </WeatherList>


      </div>
    );
  }

  handleChange = (event) => {
    this.setState({zipcode: event.target.value})
  }

  handleSubmit = (event) => {
    loadWeather(fetch, this.state.zipcode)
      .then(response =>
        this.setState(
          {
            city: response.city,
            weathers: response.weather
          }
        )
      )
    event.preventDefault();
  }
}

export default App;
