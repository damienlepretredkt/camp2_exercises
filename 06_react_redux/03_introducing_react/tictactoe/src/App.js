import React, { Component } from 'react';
import './App.css';
import Board from './Board';
import WINNING_COORDINATES from './solutions';
import { flattenArray, isNotNull } from './tools';

class App extends Component {



  constructor(props) {
    super(props);
    const randomPlayer = ["X", "O"][Math.round(Math.random())]
    this.state = {
      currentPlayer: randomPlayer,
      value: '',
      title: `Tic Tac Toe , ${randomPlayer} now it's time to play`,
      gameState: {
        a: Array(3).fill(null),
        b: Array(3).fill(null),
        c: Array(3).fill(null)
      }
    }
    this.handleChange = this.handleChange.bind(this);
    this.handleCellClick = this.handleCellClick.bind(this)
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">{this.state.title}</h1>
        </header>

          <div className="center-table">
            <Board gameState={this.state.gameState} handleCellClick={this.handleCellClick}></Board>
          </div>

        <button onClick={this.newGame}>Reset game</button>
      </div>
    );
  }

  handleChange = (event) => {
    this.setState({value: event.target.value});
  }

  handleCellClick(cellCoordinate) {

    const coordinate = this.getCoordinate(cellCoordinate, this.state.gameState);
    if (coordinate) {
      this.updateState(coordinate, this.state.gameState);
      if (this.hasWinner(this.state.gameState)) {

        this.setState({
          title: `Congratulations ${this.state.currentPlayer}, you won! ＼(＾O＾)／)`
        })

      } else if (this.gameIsFinished(this.state.gameState)) {

        this.setState({
          title: `Looks like it's a tie. Thanks for playing! ¯\\_(ツ)_/¯`,
        })


      } else {
        this.nextPlayer();
      }
    } else {
      this.setState({
        title: 'This is not a valid move'
      })
    }
  }

  hasWinner(gameState) {
  const isWinningLine = (line) => {
    const pattern =
      line
        .map((coordinate) => gameState[coordinate.letter][coordinate.digit])
        .join("");

    return pattern === "XXX" || pattern === "OOO";
  };

  return WINNING_COORDINATES.some(isWinningLine);
}

  getCoordinate(input, gameState) {
    const letter = input[0];
    const digit = input[1] - 1;

    if (gameState[letter] && gameState[letter][digit] === null) {
      return { letter: letter, digit: digit };
    } else {
      return null;
    }
  }

  updateState(coordinate, gameState) {
    const line = gameState[coordinate.letter];

    line[coordinate.digit] = this.state.currentPlayer;
  }

  nextPlayer() {
    if (this.state.currentPlayer === "X") {
      this.setState({
        currentPlayer: "O",
        title: "Tic Tac Toe , O now it's time to play",
        value: ''
      });
    } else {
      this.setState({
        currentPlayer: "X",
        title: "Tic Tac Toe, X now it's time to play",
        value: ''
      });
    }
  }

  gameIsFinished(newGameState) {
    const allValues = flattenArray(Object.values(newGameState));

    return allValues.every(isNotNull);
  }

  newGame = () => {
    const newGame = {
      a: Array(3).fill(null),
      b: Array(3).fill(null),
      c: Array(3).fill(null)
    }

    const newPlayer = ["X", "O"][Math.round(Math.random())];

    this.setState({
      gameState: newGame,
      currentPlayer: newPlayer,
      title: `Tic Tac Toe , ${newPlayer} now it's time to play`
    })
  }

}



export default App;
