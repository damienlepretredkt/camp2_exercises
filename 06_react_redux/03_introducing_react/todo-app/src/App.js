import React, { Component } from 'react';
import './App.css';
import { ListGroup, ListGroupItem, FormGroup, FormControl, Checkbox, Button } from 'react-bootstrap';



class App extends Component {

  constructor(props) {
    super(props);
    this.state = {
      todos: [],
      value: '',
      checkboxChecked: true
    }
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Welcome to TodoApp</h1>
        </header>
        <div className="container">

        <form onSubmit={this.handleSubmit}>
  				<FormGroup controlId="formBasicText">
  					<FormControl
  						type="text"
  						value={this.state.value}
  						placeholder="Enter todo name"
  						onChange={this.handleChange}/>
  				</FormGroup>
  			</form>

        <ListGroup>
          <span className="head-taks-group">Tasks Open</span>
          {this.state.todos.filter(todo => todo.status === 'open').map((todo, index) =>
          <ListGroupItem key={todo.name}>
              <Checkbox onChange={() => this.handleCheck(todo.name, 'done')}>
                <span>{todo.name}</span>
              </Checkbox>
          </ListGroupItem>)}
        </ListGroup>

        <ListGroup>
          <span className="head-taks-group">Tasks done</span>
          {this.state.todos.filter(todo => todo.status === 'done').map((todo, index) =>
          <ListGroupItem key={todo.name}>
              <Checkbox checked={this.state.checkboxChecked} onChange={() => this.handleCheck(todo.name, 'open')}>
              <span className="todo-done">{todo.name}</span>
              </Checkbox>
          </ListGroupItem>)}
        </ListGroup>

        <div>
          <Button bsStyle="primary" onClick={() => this.removeDone()}>Delete done tasks</Button>
        </div>
        </div>
      </div>
    );
  }

  handleChange = (event) => {
    this.setState({value: event.target.value});
  }

  handleCheck = (selectedTodo, status) => {
    const todoIndex = this.state.todos.findIndex((todo) => todo.name === selectedTodo)
    const newTodo = this.state.todos[todoIndex]
    newTodo.status = status;
    const newTodos = [...this.state.todos]
    newTodos.splice(todoIndex, 1)
    newTodos.push(newTodo);

    this.setState({todos: [...newTodos]})
  }

  removeDone = () => {
    const cleanTodoList = this.state.todos.filter(todo => todo.status === 'open');
    this.setState({todos: [...cleanTodoList]})
  }

  handleSubmit = (event) => {
    const newTodo = {
      name: this.state.value,
      status: 'open'
    }
    this.setState({todos: [...this.state.todos, ...[newTodo]]})
    event.preventDefault();
  }
}

export default App;
