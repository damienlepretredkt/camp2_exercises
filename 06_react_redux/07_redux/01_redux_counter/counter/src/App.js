import React from 'react';
import logo from './logo.svg';
import './App.css';
import TitleComponent from './TitleComponent'
import CounterComponent from './CounterComponent'


function App(props) {

  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <h1 className="App-title">Welcome to React</h1>
      </header>
      <div className="container">
        <div className="counter">
          <TitleComponent></TitleComponent>
          <CounterComponent></CounterComponent>
        </div>
      </div>
    </div>
  );
}

export default App;
