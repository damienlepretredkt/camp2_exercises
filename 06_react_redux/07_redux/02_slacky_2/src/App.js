import React, {Component} from "react";
import "./App.css";
import LoginComponent from "./Login";
import ChatComponent from "./Chat";

import { connect } from "react-redux";
import {
  BrowserRouter as Router,
  Route,
  Switch
} from 'react-router-dom'

import store from './store'

import { replace } from 'react-router-redux'

class App extends Component {

  componentDidMount() {
    this.props.username ? store.dispatch(replace('/chats')) : store.dispatch(replace('/login'))
  }

  render() {
    return (
      <Router>
          <ConnectedSwitch>
            <Route path="/login" component={LoginComponent}/>
            <Route exact path="/chats" component={ChatComponent} />
            <Route path="/chats/:id" component={ChatComponent} />
          </ConnectedSwitch>
      </Router>
    );
  }
}

function mapDispatchToProps(dispatch) {
  return {
    handleIncomingMessage: (messages) =>
      dispatch({type: "INCOMINGMESSAGES", messages: messages})

  }
}

function mapStateToProps(state) {
  return {
    userName: state.login.userName,
  }
}

const ConnectedSwitch = connect(state => ({
  location: state.router.location
}))(Switch)




const AppComponent = connect(mapStateToProps, mapDispatchToProps)(App);
export default AppComponent;
