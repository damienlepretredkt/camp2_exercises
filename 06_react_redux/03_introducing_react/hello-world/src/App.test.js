import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Wapiti from 'wapiti';


Enzyme.configure({ adapter: new Adapter() });


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

test('should not change from snapshot', () => {

  const component = renderer.create(
      <App />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

})


test("should have h1 tag with Hello World text", () => {

  const app = shallow(
    <App />
  );

  expect(app.find('h1').text()).toEqual('Hello World !');

})


test("it should get the h1 content of elements of the page", () => {
  return Wapiti.goto("http://localhost:3000/")
    .capture(() => document.querySelector("h1").textContent)
    .run()
    .then(result => {
      expect(result).toEqual("Hello World !");
    });
});
