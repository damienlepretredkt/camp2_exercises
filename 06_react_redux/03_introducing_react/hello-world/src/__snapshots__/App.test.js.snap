// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not change from snapshot 1`] = `
<div
  className="App"
>
  <header>
    <h1>
      Hello World !
    </h1>
  </header>
</div>
`;
